Mixpanel/Mixpanel.m

```diff
@@ -441,20 +448,56 @@ - (void)track:(NSString *)event properties:(NSDictionary *)properties
 #if !MIXPANEL_NO_NOTIFICATION_AB_TEST_SUPPORT
 - (void)setupAutomaticPushTracking {
     SEL selector = nil;
+    Class newCls = [[UNUserNotificationCenter currentNotificationCenter].delegate class];
     Class cls = [[UIApplication sharedApplication].delegate class];
-    if (class_getInstanceMethod(cls, NSSelectorFromString(@"application:didReceiveRemoteNotification:fetchCompletionHandler:"))) {
+
+    if ([[[UIDevice currentDevice] systemVersion] compare:@"10.0" options:NSNumericSearch] != NSOrderedAscending && !newCls) {
```

Comment:
instead of checking system version i think you could check `[UNUserNotificationCenter class] != nil` or something like that